mcpx/client.py

```diff
@@ -388,14 +398,26 @@ def list_installs(self) -> Iterator[Servlet]:
         List all installed servlets, this will make an HTTP
         request each time
         """
-        url = self.endpoints.installations
+        url = self.endpoints.installations(self.config.profile)
         self.logger.info(f"Listing installed mcp.run servlets from {url}")
+        headers = {}
+        if self.install_cache.last_update is not None:
+            modified = self.install_cache.last_update.strftime(
+                "%a, %d %b %Y %H:%M:%S GMT"
+            )
+            headers["if-modified-since"] = modified
         res = requests.get(
             url,
+            headers=headers,
             cookies={
                 "sessionId": self.session_id,
             },
         )
+        res.raise_for_status()
```

Comment:
You can also use the `Date` header of the last response as the value of `If-Modified-Since`, which might simplify some of the timekeeping logic here.

Reply:
good call, storing the result of the `Date` response header is a little cleaner, but also makes it more clear what's going on there.